### PR TITLE
refactor: Extract functions to parent

### PR DESCRIPTION
**Pull request by rafaeelaudibert**
These functions will be reused in `stats_table.py`, so it makes sense to extract them one level higher. There's some extra complication on how we handle `self.query.compare` because `stats_table.py` always wants to compare, so we need that to be a parameter

This is just refactoring work to allow us to better filter by `ConversionGoal` on `stats_table.py`. If anything, this is just better code (to some definition of better)

### CODE REVIEW

**Comment by robbie-c** [comment]
`WebTopClicksQuery` can be deleted if doing so would make your life easier, it's not used any more

**Comment by rafaeelaudibert** [comment]
> `WebTopClicksQuery` can be deleted if doing so would make your life easier, it's not used any more

@robbie-c It's not complicating me right now, but I'm always happy to remove code :)

**Comment by robbie-c** [comment]
I was just looking at the mypy errors ;)

**Comment by rafaeelaudibert** [comment]
> I was just looking at the mypy errors ;)

@robbie-c Oh, duh :). Might remove it then, thx

File: frontend/__snapshots__/scenes-other-onboarding--onboarding-billing--light.png
Comment (rafaeelaudibert) [inline frontend/__snapshots__/scenes-other-onboarding--onboarding-billing--light.png:1]: Ahn, looks very wrong, hog is gone 🔥, I will make it regenerate this once Depot is back online

**Comment by rafaeelaudibert** [comment]
@robbie-c tests are passing now :)

**Review by robbie-c — APPROVED** [review]

**Pull request merged**